src/main/java/strategy/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,16 @@
+package strategy.handler;
+
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public Map<String, Integer> getCurrentBalanceByFruit(List<FruitTransaction> fruitTransactions,
+                                                         int index) {
+        currentBalanceByFruit.put(fruitTransactions.get(index).getFruit(),
+                fruitTransactions.get(index).getQuantity());
+        return currentBalanceByFruit;
+    }
```

Comment:
```suggestion
    @Override
    public void handle(FruitTransaction fruitTransaction) {
        Storage.fruits.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
    }
```

Reply:
Hope my solution and structure is suitable for this task

Reply:
fixed